### ISSUE TITLE
Could we get the ability to set the rootQueryPath per search?

### ISSUE THREAD

**Issue by Robert-W** [Asymmetrik/kyruus-node-sdk#8]
Currently the root query path always returns the following:

`/pm/${this.version}/${this.source}/`

With the availability API, it has a different URL, something like this: 

`availability/<version>/<method>/<source>`.  We can specify pathname in the search function but that is just appended to the root path.  Would be nice if we could override the root query path either when calling search or as part of the query builder process

**Comment by andrewmarcus** [comment]
Makes sense to me.  The SDK was built around us only having access to that one API.  I would assume the version and source are the same?

**Comment by Robert-W** [comment]
Unfortunately the version is not, the source is though. The availability api is only on v2 while the /pm/ base is on v8. I was thinking of adding a function to override the base for a single query which would a simple string, maybe something like this:
```javascript
const query = kyruus.query()
  .param('provider_id', 123456)
  // path could set some internal variable like _overridePath
  .path('/availability/v2/get/<source>');

query.search();

// Change how root query path works in kyruus-sdk.js
__rootQueryPath () => this._overridePath ? this._overridePath : <originalRootPath>;
```

I can submit a PR later today or early tomorrow to support this
